scenedetect/cli/context.py

```diff
@@ -204,7 +204,7 @@ def _generate_images(self, scene_list, video_name,
         else:
             middle_images = self.num_images - 2
             for i, (start_time, end_time) in enumerate(scene_list):
-                timecode_list[i].append(start_time)
+                timecode_list[i].append(start_time + self.image_frame_margin)
```

Comment:
Would prefer bound checks on the timecodes being used to make sure they do not exceed the frame boundaries.  Specifically, if the calculated start/end times would exceed the half-way point, I think that should be defined behavior - what that behavior should be, though, I'm not 100% sure on, so any feedback or opinions you have would be welcome.

Reply:
Also I know I'm essentially asking you to do work for free, so if you are unable to support any of the requested changes I've made on this PR or any other ones, please don't hesitate to let me know, and I'll take over the rest of the PR accordingly.

Very pleased with the quality of your commits thus far though (very small and targeted changes), sorry I don't have more time to actively support them as much as I used to.  Thank you again for your PRs and issue requests, these are definitely things I want to add in v0.5.2 if possible (hurray for non-breaking changes :))!

Reply:
See what you think about a393efc.   It's a pretty significant rewrite of the code that builds the timecode list for images, but I think it accomplishes what you want by dividing each scene's frames into `--num-images` chunks and then selecting the appropriate frame from each.

The `min` and `max` make sure that the frame for the first and last images never go past the frames for the middle images.  Lightly tested with some crazy high values of `--image-frame-margin` and it seems to work as expected.

Reply:
Awesome, have you tested with values of 0 and 1 by any chance?  Also I'd like to write some unit tests for this eventually as well, but won't let that keep this from being merged.

Would like to do some testing on this myself before merging, but at a first glance your implementation looks a lot cleaner than mine, thank you! 😄 👍